boards/components/src/fxos8700.rs

```diff
@@ -6,54 +6,72 @@
 //! -----
 //! ```rust
 //! let fxos8700 = components::fxos8700::Fxos8700Component::new(mux_i2c, PinId::AdB1_00.get_pin().as_ref().unwrap())
-//!    .finalize(());
+//!    .finalize(components::fxos8700_component_static!());
 //!
 //! let ninedof = components::ninedof::NineDofComponent::new(board_kernel)
-//!    .finalize(components::ninedof_component_helper!(fxos8700));
+//!    .finalize(components::ninedof_component_static!(fxos8700));
 //! ```
 
 // Based on the component written for sam4l by:
 // Author: Philip Levis <pal@cs.stanford.edu>
 // Last modified: 6/03/2020
 
-use capsules::fxos8700cq;
+use capsules::fxos8700cq::Fxos8700cq;
 use capsules::virtual_i2c::{I2CDevice, MuxI2C};
 
 use kernel::component::Component;
 
+use core::mem::MaybeUninit;
 use kernel::hil;
 use kernel::hil::gpio;
-use kernel::static_init;
+
+#[macro_export]
+macro_rules! fxos8700_component_static {
+    () => {{
+        let i2c_device = kernel::static_buf!(capsules::virtual_i2c::I2CDevice);
+        let buffer = kernel::static_buf!([u8; capsules::fxos8700cq::BUF_LEN]);
+        let fxo = kernel::static_buf!(capsules::fxos8700cq::Fxos8700cq<'static>);
+
+        (i2c_device, buffer, fxo)
+    };};
```

Comment:
```suggestion
    }};
```
Does this not generate a warning here?

Reply:
I'm confused what is causing the warning, or not causing the warning in the case of capsules. I've seen other PRs that _add_ the semicolon to fix something, so I'm not really sure what the right thing to do is.

Reply:
I thought it might be for when the macro returns something, like in this case and in https://github.com/tock/tock/pull/3245 . But if there's no warnings I'm not sure either...🤔